Source/CarthageKit/Project.swift

```diff
@@ -913,9 +913,11 @@ public final class Project { // swiftlint:disable:this type_body_length
 						continue
 					}
 
-					return Result(at: dependencyCheckoutURL, attempt: {
+					if let error = Result(at: dependencyCheckoutURL, attempt: {
```

Comment:
Mostly, this is just a stopgap before a larger refactor, which would look more like the following:

##### Preview with indenting changes removed:

```diff
 		// (which could be outside `CarthageProjectCheckoutsPath`).
 		if names.isEmpty { return .success() } // swiftlint:disable:this single_line_return
 
+		return Result(at: dependencyCheckoutsURL, attempt: {
 		do {
-			try fileManager.createDirectory(at: dependencyCheckoutsURL, withIntermediateDirectories: true)
+			try fileManager.createDirectory(at: $0, withIntermediateDirectories: true)
 		} catch let error as NSError {
 			if !(error.domain == NSCocoaErrorDomain && error.code == NSFileWriteFileExistsError) {
-				return .failure(.writeFailed(dependencyCheckoutsURL, error))
+				throw error
 			}
 		}
 
 		for name in names {
-			let dependencyCheckoutURL = dependencyCheckoutsURL.appendingPathComponent(name)
+			let dependencyCheckoutURL = $0.appendingPathComponent(name)
 			let subdirectoryPath = (carthageProjectCheckoutsPath as NSString).appendingPathComponent(name)
 			let linkDestinationPath = relativeLinkDestination(for: dependency, subdirectory: subdirectoryPath)
 
-			let dependencyCheckoutURLResource = try? dependencyCheckoutURL.resourceValues(forKeys: [
-				.isSymbolicLinkKey,
-				.isDirectoryKey,
-			])
+			let dependencyCheckoutURLCheck = {
+				try dependencyCheckoutURL.copyResource(bridgingFromType: CFBoolean.self, forIdentifierKey: $0) == .some(true)
+			}
 
-			if dependencyCheckoutURLResource?.isSymbolicLink == true {
+			if try dependencyCheckoutURLCheck(kCFURLIsSymbolicLinkKey) {
 				_ = dependencyCheckoutURL.path.withCString(Darwin.unlink)
-			} else if dependencyCheckoutURLResource?.isDirectory == true {
+			} else if try dependencyCheckoutURLCheck(kCFURLIsDirectoryKey) {
 				// older version of carthage wrote this directory?
 				// user wrote this directory, unaware of the precedent not to circumvent carthage’s management?
 				// directory exists as the result of rogue process or gamma ray?
 
 				// TODO: explore possibility of messaging user, informing that deleting said directory will result
 				// in symlink creation with carthage versions greater than 0.20.0, maybe with more broad advice on
-				// “from scratch” reproducability.
+				// “from scratch” reproducibility.
 				continue
 			}
 
-			if let error = Result(at: dependencyCheckoutURL, attempt: {
-				try fileManager.createSymbolicLink(atPath: $0.path, withDestinationPath: linkDestinationPath)
-			}).error {
-				return .failure(error)
-			}
+			try fileManager.createSymbolicLink(atPath: dependencyCheckoutURL.path, withDestinationPath: linkDestinationPath)
 		}
-
-		return .success()
+		})
 	}
```